src/medCore/gui/viewContainers/medViewContainer.cpp

```diff
@@ -829,3 +859,115 @@ void medViewContainer::removeColorIndicator(QColor color)
 {
     d->poolIndicator->removeColorIndicator(color);
 }
+
+/**
+ * @brief saves the scene in a XML file
+ * Saves views (all layers), and toolboxes parameters
+ * 	expected tree is as follow:
+ *  workingDir (user-defined)
+ * 		|-viewID0
+ * 			|-mapping.xml
+ * 			|-layerID01.xml
+ * 			|-layerID02.xml
+ * 			|-data_file (as saved by the dedicated writer)
+ */
+QString medViewContainer::saveScene()
+{
+    QString dirPath = QFileDialog::getExistingDirectory(this, tr("Open Directory"),"/home",QFileDialog::ShowDirsOnly);
+    QDir workingDir(dirPath);
+    QDomDocument doc("xml");
+    QDomElement root=doc.createElement("scene");
+    doc.appendChild(root);
+
+    QString subDirName;
+
+    medAbstractLayeredView* layeredView=dynamic_cast<medAbstractLayeredView*>(view());
+    if (layeredView!=0)
+    {
+        subDirName="view"+QUuid::createUuid ().toString();
+        if(!workingDir.mkdir(subDirName))
+            qWarning() << " failed to create new directory: "<<subDirName;
+        workingDir.cd(subDirName);
+        QString generatedPath=workingDir.canonicalPath()+"/mapping.xml";
+        QDomElement layeredViewInfo=doc.createElement("layeredView");
+        layeredViewInfo.setAttribute("path",subDirName+"/mapping.xml");
+        layeredViewInfo.setAttribute("id",layeredView->identifier());
+        root.appendChild(layeredViewInfo);
+        layeredView->write(generatedPath);
+        workingDir.cdUp();
+    }
+    QString generatedPath=workingDir.canonicalPath()+"/globalMapping.xml";
+
+    QFile file(generatedPath);
+    if (!file.open(QIODevice::WriteOnly | QIODevice::Text | QIODevice::Truncate))
+        return "";
+
+    QTextStream out(&file);
+    out << doc.toString();
+    QString path = dirPath + "/" + subDirName;
+    return path;
+}
+
+void medViewContainer::loadScene()
+{
+    //parsing the XML file describing the scene,
+    QString fileName = QFileDialog::getOpenFileName(this, tr("Open File"),"/home",tr("XML files (*.xml)"));
+    QFile file(fileName);
+    QFileInfo fileInfo(file);
+    QDir workingDir=fileInfo.dir();
+    if(!file.open(QIODevice::ReadOnly))
+    {
+         qWarning() << "failed to open file "<<fileName;
```

Comment:
This is an interesting one.

On the one hand, if the xml fails to load we should probably warn the user - after all, they're not getting the behaviour they expect, and they won't know why.

On the other hand, if they manually change the xml file and corrupt it, then they're a moron and are doing things we do not support. We can't be expected to predict every single stupid thing a user would do and protect them from their own idiocy.

However, from a practical point of view, it doesn't really cost us much to tell the user that s/he's done something wrong. As long as the encoding is correct, then the only time we'd hit an error state would be if the user had done something dumb. Therefore, this seems more like a user error for me. It will also be useful for when we screw up the code, but primarily one would expect to see it when some over-confident user tries to be clever and ends up embarrassing themselves.